lib/ACL/ACLManager.php

```diff
@@ -98,13 +99,31 @@ private function getRules(array $paths, bool $cache = true): array {
 	 * @return string[]
 	 */
 	private function getRelevantPaths(string $path): array {
-		$paths = [$path];
+		$paths = [];
+		$fromTrashbin = str_starts_with($path, '__groupfolders/trash/');
+		if ($fromTrashbin) {
+			/* Exploded path will look like ["__groupfolders", "trash", "1", "folderName.d2345678", "rest/of/the/path.txt"] */
+			[,,$groupFolderId,$rootTrashedItemName] = explode('/', $path, 5);
+			$groupFolderId = (int)$groupFolderId;
+			/* Remove the date part */
+			$separatorPos = strrpos($rootTrashedItemName, '.d');
+			$rootTrashedItemDate = (int)substr($rootTrashedItemName, $separatorPos + 2);
+			$rootTrashedItemName = substr($rootTrashedItemName, 0, $separatorPos);
+		}
 		while ($path !== '') {
+			$paths[] = $path;
 			$path = dirname($path);
+			if ($fromTrashbin && ($path === '__groupfolders/trash')) {
+				/* We are in trash and hit the root folder, continue looking for ACLs on parent folders in original location */
+				$trashItemRow = $this->trashManager->getTrashItemByFileName($groupFolderId, $rootTrashedItemName, $rootTrashedItemDate);
+				$path = dirname('__groupfolders/' . $trashItemRow['folder_id'] . '/' . $trashItemRow['original_location']);
```

Comment:
Just to be clear that we are only calling `getTrashItemByFileName` for the `original_location`.

```suggestion
				$path = dirname('__groupfolders/' . $groupFolderId . '/' . $trashItemRow['original_location']);
```